taskpools/channels_spsc_single.nim

```diff
@@ -144,21 +133,23 @@ when isMainModule:
           discard
         echo "Sender sent: ", val
 
+  import primitives/allocs
   proc main() =
     echo "Testing if 2 threads can send data"
     echo "-----------------------------------"
 
     var threads: array[2, Thread[ThreadArgs]]
-    var chan = cast[ptr ChannelSPSCSingle](c_calloc(1, csize_t sizeof(ChannelSPSCSingle)))
-    chan[].initialize(itemSize = sizeof(int))
+    var chan = tp_allocAligned(
+      ChannelSPSCSingle[int], sizeof(ChannelSPSCSingle[int]), 64)
+    zeroMem(chan, sizeof(ChannelSPSCSingle[int]))
```

Comment:
This patterns seems to be repeated in few places. It would be nice to introduce a simpler helper called `tp_alloc0Aligned`?

Reply:
I thought about that too but it would significantly grow the scope of this PR since all of the library should ideally be updated

Reply:
also, in some cases the library opts to not zero the full range of allocated bytes, so a "0" helper could not universally be applied, again pointing in the direction of a full review rather than a drive-by fix in this PR